packages/entities/entities-shared/src/composables/useErrors.ts

```diff
@@ -70,7 +70,92 @@ export default function useErrors() {
     return error.message || t('errors.unexpected')
   }
 
+  /**
+ * Receive an error and return the error fields from the payload
+ * @param {string|Error|object} error - error string
+ * @returns {Object} Object containing message and fields
+ */
+  const getErrorFieldsFromError = (error: any): { messages: string[]; fields: Array<{ field: string; message: string }> } => {
+    if (!error) {
+      return {
+        messages: [t('errors.unexpected')],
+        fields: [],
+      }
+    }
+
+    if (error?.response?.status === 401) {
+      console.warn('getErrorFieldsFromError: User auth token is expired or missing, returning empty messages.')
+      return {
+        messages: [],
+        fields: [],
+      }
+    }
+
+    let messages: string[] = [t('errors.unexpected')]
+    const fields: Array<{ field: string; message: string }> = []
+
+    if (error?.response?.data) {
```

Comment:
is it possible at all to add comments anywhere explaining what/how the data is being transformed?